conformance/lang/expressions/trap_expr.balt

```diff
@@ -0,0 +1,351 @@
+Test-Case: output
+Description: Test the result of the trap expression being `e`,
+             when evaluation of the expression completes abruptly with panic with associated value `e`.
+Labels: error-type, function-call-expr, function-defn, optional-type, panic, trap-expr
+
+function foo() {
+    panic error("foo panics");
+}
+
+function init() {
+    error? e1 = trap foo();
+    io:println(e1); // @output error("foo panics")
+}
+
+Test-Case: output
+Description: Test the result of the trap expression being `v`,
+             when evaluation of the expression does not complete abruptly with panic.
+Labels: error-type, function-call-expr, function-defn, int, nil-literal, nil-type, optional-type, trap-expr, union-type, variable-reference-expr
+
+function init() {
+    int? nilVal = ();
+    int|error? e = trap nilVal;
+    io:println(e is ()); // @output true
+}
+
+Test-Case: error
+Description: If type of expr is T, then type of trap expr is T|error.
+Labels: error-type, int, local-init-var-decl-stmt, multiplicative-expr, nil-literal, optional-type, string, trap-expr, union-type, variable-reference-expr
+
+function errorFunction() {
+    int zero = 0;
+    int? nilVal = ();
+    int _ = trap 2 / zero; // @error incompatible types: expected 'int', found '(int|error)'
+    string|error? x = trap nilVal; // @error incompatible types: expected '(string|error)?', found 'int?'
+    int _ = trap 2; // @error incompatible types: expected 'int', found '(int|error)'
+    string _ = trap "str"; // @error incompatible types: expected 'string', found '(string|error)'
+    float _ = trap 2.3f; // @error incompatible types: expected 'float', found '(float|error)'
+}
+
+Test-Case: error
+Description: If type of expr is T, then type of trap expr is T|error.
+Fail-Issue: ballerina-platform/ballerina-lang#36472
+Labels: error-type, int, local-init-var-decl-stmt, multiplicative-expr, trap-expr, variable-reference-expr
```

Comment:
`local-init-var-decl-stmt` is part of the minimal subset. Why are we specifically mentioning it here?

Reply:
So is `variable-reference-expr`.

Reply:
Removed them